web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -138,6 +144,27 @@ export class LayerDialogComponent implements OnDestroy {
     );
   }
 
+  deleteQuestion(index: number) {
+    const dialogRef = this.confirmationDialog.open(
+      ConfirmationDialogComponent,
+      {
+        maxWidth: '500px',
+        data: {
+          title: 'Warning',
+          message:
+            'Are you sure you wish to delete this field? Any associated data will be lost. This cannot be undone.',
+        },
+      }
+    );
+
+    dialogRef.afterClosed().subscribe(dialogResult => {
```

Comment:
How does the subscription get disposed? Don't we need to keep it around in order to delete it when the UI is destroyed?

Reply:
No need to unsubscribe as it auto completes itself. Check this out [Refs](https://github.com/angular/components/blob/ae41a0ad69ca26c600f0f56c68dd5a1c102d4f1f/src/lib/dialog/dialog-ref.ts#L75)